pkg/apis/networking/v1alpha1/clusteringress_types.go

```diff
@@ -0,0 +1,379 @@
+	// will be forwarded to.
+	Splits []ClusterIngressBackendSplit `json:"splits"`
+
+	// AppendHeaders allow specifying additional HTTP headers to add
+	// before forwarding a request to the destination service.
+	//
+	// NOTE: This differs from K8s Ingress which doesn't allow header appending.
+	// +optional
+	AppendHeaders map[string]string `json:"appendHeaders,omitempty"`
+
+	// Timeout for HTTP requests.
+	//
+	// NOTE: This differs from K8s Ingress which doesn't allow setting timeouts.
+	// +optional
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+
+	// Retry policy for HTTP requests.
+	//
+	// NOTE: This differs from K8s Ingress which doesn't allow retry settings.
+	// +optional
+	Retries *HTTPRetry `json:"retries,omitempty"`
+}
+
+// HTTPRetry describes the retry policy to use when a HTTP request fails.
+type HTTPRetry struct {
+	// Number of retries for a given request.
+	Attempts int `json:"attempts"`
+
+	// Timeout per retry attempt for a given request. format: 1h/1m/1s/1ms. MUST BE >=1ms.
+	PerTryTimeout *metav1.Duration `json:"perTryTimeout"`
+}
+
+// ClusterIngressBackend describes all endpoints for a given service and port.
+type ClusterIngressBackend struct {
+	// Specifies the namespace of the referenced service.
+	//
+	// NOTE: This differs from K8s Ingress to allow routing to different namespaces.
+	ServiceNamespace string `json:"serviceNamespace"`
+
+	// Specifies the name of the referenced service.
+	ServiceName string `json:"serviceName"`
+
+	// Specifies the port of the referenced service.
+	ServicePort intstr.IntOrString `json:"servicePort"`
+}
+
+// ClusterIngressBackend describes all endpoints for a given service and port.
+type ClusterIngressBackendSplit struct {
+	// Specifies the backend receiving the traffic split.
+	Backend *ClusterIngressBackend `json:"backend"`
+
+	// Specifies the split percentage, a number between 0 and 100.  If
+	// only one split is specified, we default to 100.
+	//
+	// NOTE: This differs from K8s Ingress to allow percentage split.
+	Percent int `json:"percent,omitempty"`
+}
+
+// IngressStatus describe the current state of the ClusterIngress.
+type IngressStatus struct {
+	// +optional
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty"`
+	// LoadBalancer contains the current status of the load-balancer.
+	// +optional
+	LoadBalancer *LoadBalancerStatus `json:"loadBalancer,omitempty"`
+}
+
+// LoadBalancerStatus represents the status of a load-balancer.
+type LoadBalancerStatus struct {
+	// Ingress is a list containing ingress points for the load-balancer.
+	// Traffic intended for the service should be sent to these ingress points.
+	// +optional
+	Ingress []LoadBalancerIngress `json:"ingress,omitempty" protobuf:"bytes,1,rep,name=ingress"`
+}
+
+// LoadBalancerIngress represents the status of a load-balancer ingress point:
+// traffic intended for the service should be sent to an ingress point.
+type LoadBalancerIngress struct {
```

Comment:
Since this only appears in Status, maybe `LoadBalancerIngressStatus`, since `IngressStatus` is already taken?